corehq/apps/reports/static/reports/js/bootstrap3/hq_report.js

```diff
@@ -67,8 +67,8 @@ hqDefine("reports/js/bootstrap3/hq_report", [
                             e.preventDefault();
                             if (self.isExportAll) {
                                 $.ajax({
-                                    url: getReportBaseUrl("export"),
-                                    data: getReportParams(undefined),
+                                    url: self.getReportBaseUrl("export"),
```

Comment:
Looking at the [initializations](https://github.com/dimagi/commcare-hq/blob/66d0a8240d3f962b08e6693f8bcae08eac81ba78/corehq/apps/reports/static/reports/js/bootstrap3/hq_report.js#L42-L44) of these functions...were they previously relying on variable hoisting to default to the functions that are defined further down in the file?

Reply:
Yes, exactly, they were using the functions defined further down. And not just defaulting, `self.getReportBaseUrl()` and `self.getReportParams()` were ignored. Of those three functions, only `self.getReportRenderUrl()` was ever used.

Reply:
Wow. The new code is much more reasonable.